charts/infra-apps/templates/argocd.yaml

```diff
@@ -0,0 +1,29 @@
+{{ if .Values.argocd.enabled }}
+{{ template "argoconfig.application" (list . "infra-apps.argocd") }}
+---
```

Comment:
I wouldn't jam two resources into one file. Unless it's _n_ resources built from a loop of course.

Reply:
All those files (argocd.yml, cermgr.yml, ingress.yml, monitoring.yml, and rbac.yml) are very similar.

Reply:
Thanks for the feedback!

Them looking similar is one of the reasons I put both the application and as well as the project into the same file. It makes it easy to just `cat old | sed > new` when creating a new one.

I'll look if I can find a sensible way to have an application.yaml as well as a approject.yaml with some looping.

The application part is mostly a workaround until [ApplicationSets](https://github.com/argoproj-labs/applicationset) are more mature so we can leverage those. Maybe the AppProject should be removed from here letting us manage rbac/security related stuff through a different channel, but then again I already added rbac.yaml which moves some serious security concerns into this chart.

Reply:
The only way I can see some looping happen is with the `tpl` function which looks like a `exec(generated_string)` kind of pattern if we use it to generate templated templates.

Since this will be replaced with an ApplicationSet once that is possible, I'm going to keep this like it is. I hope keeping it this way will make it easier to migrate to such an ApplicationSet down the line.